File: codegen/results/v001/qwen3.5_4b-nvfp4_nothink/20260409_091753/python/004_json_filter/solution.py
import json

with open("input/data.json", "r") as f:
    data = json.load(f)

filtered = [record for record in data if record.get("active") and record.get("age", 0) >= 30]
filtered.sort(key=lambda x: x["name"])

print(json.dumps(filtered, ensure_ascii=False))

File: codegen/results/v001/qwen3.5_4b-nvfp4_nothink/20260409_091753/typescript/004_json_filter/solution.ts
import * as fs from 'fs';

const data = JSON.parse(fs.readFileSync('input/data.json', 'utf8'));

// Filter records: active === true AND age >= 30
const filteredData = data.filter((record) => record.active && record.age >= 30);

// Sort by name ascending
filteredData.sort((a, b) => a.name.localeCompare(b.name));

// Output as JSON to stdout
console.log(JSON.stringify(filteredData, null, 2));